packages/docs-page/server/get-paths-from-nav-data.ts

```diff
@@ -18,13 +18,16 @@ export function getPathArraysFromNodes(navNodes: NavNode[]) {
 export function getPathsFromNavData(
   navDataResolved: NavNode[],
   paramId: string = DEFAULT_PARAM_ID
-) {
+): {
+  params: Record<string, string[]>
+}[] {
   //  Transform navigation data into path arrays
   const pagePathArrays = getPathArraysFromNodes(navDataResolved)
-  // Include an empty array for the "/" index page path
-  const allPathArrays = [[]].concat(pagePathArrays)
-  const paths = allPathArrays.map((p) => ({ params: { [paramId]: p } }))
-  return paths as {
-    params: Record<string, string[]>
-  }[]
+  // Ensure we include an empty array for the "/" index page path
+  // (may be included in nav-data, eg for Terraform, or may not, eg for all other sites)
+  const hasIndexPage = pagePathArrays.filter((p) => p == []).length > 0
+  if (!hasIndexPage) pagePathArrays.unshift([])
```

Comment:
Updated this slightly, in some contexts (Terraform 👀) consumers may have added the index page to sidenav data.